endpoints/s3/src/putobject.cpp

```diff
@@ -476,12 +476,56 @@ void irods::s3::actions::handle_putobject(
 					auto header = parser_message.find("x-amz-decoded-content-length");
 					if (header != parser_message.end()) {
 						part_size = boost::lexical_cast<uint64_t>(parser_message["x-amz-decoded-content-length"]);
+
+						// Make sure someone hasn't previously uploaded this part with a different part size.
+						// If so we can't handle that and have to reject the call.
+						if (part_shmem::part_size_map[upload_id].find(part_number_int) !=
+						    part_shmem::part_size_map[upload_id].end()) {
+							auto old_part_size = part_shmem::part_size_map[upload_id][part_number_int];
+							if (old_part_size != part_size) {
+								// reject this
+								logging::error(
+									"{}: Upload ID {} - part_number {} was uploaded a second time with a different "
+								    "part size. "
+									"Rejecting this request",
```

Comment:
Is it worth logging the old part size and the new part size in the message? The answer to this question may very well be "no".

Reply:
can't hurt i think.  at this point... it's a very interesting case to see more informaiton.

Reply:
Done